pkg/api/index.go

```diff
@@ -234,7 +234,21 @@ func (hs *HTTPServer) getNavTree(c *models.ReqContext, hasEditPerm bool, prefs *
 	if err != nil {
 		return nil, err
 	}
-	navTree = append(navTree, appLinks...)
+
+	// When topnav is enabled we can test new information architecture where plugins live in Apps category
+	if hs.Features.IsEnabled(featuremgmt.FlagTopnav) {
+		navTree = append(navTree, &dtos.NavLink{
+			Text:        "Apps",
+			Icon:        "apps",
+			Description: "App plugins",
+			Id:          "apps",
+			Children:    appLinks,
+			Section:     dtos.NavSectionCore,
+			Url:         hs.Cfg.AppSubURL + "/apps",
+		})
+	} else {
+		navTree = append(navTree, appLinks...)
+	}
```

Comment:
**Question:** although I like this approach, what do you think we could do to add support for placing app plugins to custom places inside the nav tree? ([related Github issue](https://github.com/grafana/grafana/issues/49788))

If we would like to allow something like that, I would probably say it should rather be some kind of config option inside Grafana instead of a configuration for the plugins, so the Grafana instance keeps the control over its nav tree. What do you think?

Reply:
@leventebalogh yes, absolutely. If we had a way through grafana ini to specify what plugins go into what categories we could use that. This code here is just temporary to test putting plugins in a section category (have a follow-up PR to this where I explore how to make this work from AppRootPage and from inside Plugin pages) very tricky to make that work and be backward compatible (support plugins that are not updated to take advantage of new page layouts)